hazelcast/src/main/java/com/hazelcast/spi/OperationFactory.java

```diff
@@ -26,9 +26,8 @@
  * {@link OperationService} uses this factory to create {@link Operation}s for each partition by calling
  * {@link #createOperation()}
  * </p>
- *
  */
-public interface OperationFactory extends DataSerializable {
```

Comment:
Hmmm. In theory we can make this change since we only provide a best effort guarantee in the SPI module, but now we are going to break every implementation that is not making use of IdentifiedDataSerializable. I have no problem with the change by itself; but it should have been done at the beginning. 

@jerrinot WDYT